pkg/cli/common.go

```diff
@@ -189,6 +216,35 @@ func GetBudFlags(flags *BudResults) pflag.FlagSet {
 	return fs
 }
 
+// GetBudFlagsCompletions returns the FlagCompletions for the common bud flags
+func GetBudFlagsCompletions() commonComp.FlagCompletions {
+	flagCompletion := commonComp.FlagCompletions{}
```

Comment:
What do you mean by automatically? I can get the flags of course but the decision what completion function has to be used cannot be done automatically.

That's why I have added a unit test. This ensures that the flags are in sync with the completions.

Reply:
I think I agree with Dan, but I'll phrase the question differently: is there any way that the `commonComp.AutoCompleteXYZ` can be specified closer to the flag definition itself, in `GetBudFlags()`? Like
```go
    fs.IntVar(&flags.Jobs, "jobs", ..., commonComp.AutoCompleteInt)
```
This has to be a common-enough pattern that there must be wrapper functions for it?

Reply:
I really wish there would be one but unfortunately no. The completion logic exists only in `spf13/cobra` but the flag logic is in `spf13/pflag`. I would love to handle this in a better way. I thought about adding a wrapper function myself but there are to many flag definition functions which would need to be handled. In the podman PR I did it like this:
```golang
connectionFlagName := "connection"
fs.StringVarP(&opts.Engine.ActiveService, connectionFlagName, "c", srv, "Connection to use for remote Podman service")
_ = cmd.RegisterFlagCompletionFunc(connectionFlagName, common.AutocompleteSystemConnections)
```
But this approach would break the `GetXXXFlags()` API that's why I choose this implementation.

Reply:
Understood. Thanks for having looked at the possibility; and thanks for the tests, at least those will catch discrepancies early.